### PR TITLE
Added path information to warnings while loading PNGs

### PR DESCRIPTION
**Pull request by rcurtis**
Godot will often issue warnings about incorrect sRGB profiles for loaded images.  It does not, however, say which image/s have caused these warnings.  This PR adds the names of the images to the warnings.

### CODE REVIEW

**Comment by groud** [comment]
This really looks like a messy workaround for low gain. If such thing should be implemented it should be in the FileAccess class I think. Something like `FileAccess fa; fa.get_file_name_string(); `.

**Comment by rcurtis** [comment]
@groud Yes I think your suggestion is far more elegant.

**Comment by volzhs** [comment]
this PR has a modification `driver/png` source.
it makes hard to update png upstream.
should be careful when updating it in the future. (not just copy it from upstream)

I'm not sure @akien-mga is happy with this.

**Comment by rcurtis** [comment]
@volzhs I don't understand.  I didn't make any changes to libpng.  How would this interfere with upstream changes to it?

**Comment by volzhs** [comment]
@rcurtis oh. it's my bad. I thought wrong. your PR is good. sorry for it.

**Review by volzhs — APPROVED** [review]

File: drivers/png/image_loader_png.cpp [hunk 1]
```diff
@@ -239,7 +243,8 @@ Error ImageLoaderPNG::_load_image(void *rf_up, png_rw_ptr p_func, Image *p_image
 }
 
 Error ImageLoaderPNG::load_image(Image *p_image, FileAccess *f) {
-
+	
+	file = f;
```
Comment (akien-mga) [inline drivers/png/image_loader_png.cpp]: If something must be stored, it's simpler to just store the file path and not the whole FileAccess.
Also, please squash commits together.

**Comment by akien-mga** [comment]
> If something must be stored, it's simpler to just store the file path and not the whole FileAccess.
Also, please squash commits together.

I still think it's not the right approach to do this, even though I agree that having more information about what file emitted the warning would be useful.

Since discussion stalled, and this should be implemented in the *master* branch in priority before working on older stable branches, closing it for now.

**Pull request closed without merging**